File: examples/node/load_any_skill.ts
/**
 * Inspect a skill's compiled prompt and payloads for every provider.
 *
 * Run with:
 *   npx tsx examples/node/load_any_skill.ts code-reviewer
 */
import { promises as fs } from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";
import { load, getAdapter, availableProviders } from "../../packages/usf-ts/src/index.js";

const here = path.dirname(fileURLToPath(import.meta.url));
const repo = path.resolve(here, "../..");

async function main() {
  const name = process.argv[2] ?? "code-reviewer";
  const skill = await load(path.join(repo, "skills", `${name}.md`));
  const fixtures = JSON.parse(
    await fs.readFile(path.join(repo, "tests/fixtures/canonical_inputs.json"), "utf-8"),
  ) as Record<string, Record<string, unknown>>;
  const compiled = await skill.buildPrompt(fixtures[name] ?? {});

  console.log(`=== Skill: ${skill.name} ===`);
  console.log("[system]");
  console.log(compiled.system);
  console.log("\n[user]");
  console.log(compiled.user);

  for (const provider of availableProviders()) {
    const payload = getAdapter(provider).render(compiled);
    console.log(`\n--- ${provider} payload ---`);
    console.log(JSON.stringify(payload, null, 2).slice(0, 500));
  }
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: examples/python/load_any_skill.py
"""Load a skill, build the prompt, and inspect the payload for every provider.

Runs entirely offline — no API keys required.

Run:
    python examples/python/load_any_skill.py code-reviewer
"""
from __future__ import annotations

import json
import sys
from pathlib import Path

from usf import adapters, load

REPO = Path(__file__).resolve().parents[2]
SKILLS_DIR = REPO / "skills"
FIXTURES = json.loads(
    (REPO / "tests" / "fixtures" / "canonical_inputs.json").read_text(encoding="utf-8")
)


def main() -> int:
    name = sys.argv[1] if len(sys.argv) > 1 else "code-reviewer"
    skill = load(SKILLS_DIR / f"{name}.md")
    inputs = FIXTURES.get(name, {})
    compiled = skill.build_prompt(inputs)
    print(f"=== Skill: {skill.name} ===")
    print(f"[system]\n{compiled.system}\n")
    print(f"[user]\n{compiled.user}\n")
    for provider in adapters.available_providers():
        payload = adapters.get_adapter(provider).render(compiled)
        print(f"--- {provider} payload ---")
        print(json.dumps(payload, indent=2, ensure_ascii=False)[:500])
        print()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
